### PR TITLE
Optimization of AABB

### PR DESCRIPTION
**Pull request by TomFischer**
This PR is a result of https://github.com/ufz/ogs/issues/710. @endJunction discovered, that creating AABB objects is relative expensive. So in this PR the size of management overhead of the class `AABB<Node>` is reduced  from 304 byte to 64 byte. I am curious on the new callgraph and run time comparisons.

### CODE REVIEW

**Comment by bilke** [comment]
:+1:

File: Applications/DataExplorer/DataView/ElementTreeModel.cpp [hunk 1]
```diff
@@ -140,8 +140,8 @@ void ElementTreeModel::setMesh(MeshLib::Mesh const*const mesh)
 	_rootItem->appendChild(aabb_item);
 
 	const GeoLib::AABB<MeshLib::Node> aabb (MeshLib::MeshInformation::getBoundingBox(*mesh));
-	const MeshLib::Node& min = aabb.getMinPoint();
-	const MeshLib::Node& max = aabb.getMaxPoint();
+	auto min = aabb.getMinPoint();
```
Comment (norihiro-w) [inline Applications/DataExplorer/DataView/ElementTreeModel.cpp]: no need to write `auto&`? maybe there is no difference but `aabb.getMinPoint()` returns a reference.

Comment (TomFischer) [inline Applications/DataExplorer/DataView/ElementTreeModel.cpp]: Changed and rebased.

**Comment by norihiro-w** [comment]
:+1:

**Comment by endJunction** [comment]
One could move the default initialization of min and max points in aabb to variable declaration such that not to repeat it in init lists of the constructors.
(The mobile version of the site does not allow in-code comments.)
:fast_forward: in any case.

**Comment by endJunction** [comment]
Two general questions.
1. Why are the ids of input points used for selection of a subset? Couldn't this be done in the calling function? This would allow to switch to mathlib::point3d, or even to templatePoint.
2. Would the following test fail?
```c++
Point a = ...;
std::vector pnts({{a}});
Auto box = AABB(pnts.begin(), pnts.end());
assert(box.containsPoint(a));
```

**Comment by TomFischer** [comment]
**Answering your first bunch of questions:**
> Why are the ids of input points used for selection of a subset?

Only one constructor is offering this possibilty.

> Couldn't this be done in the calling function?

This would be possible. As far as I see this constructor is used seldom. Then it is possible to remove the corresponding constructor.

> This would allow to switch to mathlib::point3d, or even to templatePoint.

This question is unclear to me. What exactly would allow to switch to `MathLib::Point3d` and what does "switch" mean. Can you explain it?

**Answer to 2.**
I do not see a reason why the test should not run succesfully.

**Comment by endJunction** [comment]
Regarding the first: I think that selection of points and AABB algorithms should be separate. Therefore the suggestions. Would it be possible then to remove the template and work only on point3d derived objects?
Second: those who can read have definitely an advantageous position :-) . Never mind.

**Pull request merged**